core/shared/platform/common/posix/posix_thread.c

```diff
@@ -368,7 +368,11 @@ os_thread_get_stack_boundary()
 #ifdef __linux__
     if (pthread_getattr_np(self, &attr) == 0) {
         pthread_attr_getstack(&attr, (void **)&addr, &stack_size);
+#if WASM_STACK_GUARD_SIZE == 0
         pthread_attr_getguardsize(&attr, &guard_size);
+#else
+        guard_size = WASM_STACK_GUARD_SIZE;
+#endif
```

Comment:
if WASM_STACK_GUARD_SIZE is smaller than 1 page, L374 takes effect, which may be too small? How about changing to:
```
        pthread_attr_getguardsize(&attr, &guard_size);
        guard_size += WASM_STACK_GUARD_SIZE;
```

Reply:
OK, it's more suitable for powerful unix like systems.